### PR TITLE
Python 3 compatibility fixes plus Windows builds

### PR DESCRIPTION
**Pull request by stephendade**
A number of things in here:
-Fixed regression issues with the Python 2->3 move
-Fixed all unit tests
-Added TravisCI build for Python 3.5 (this is in addition to Python 2.7)
-Added Windows build (via appveyor) plus unit testing (all works! except realtime image processing)
-Now using piexif for all jpeg exif reading/writing (more compatible with Windows and/or Python3)

### CODE REVIEW

File: cuav/image/imageutil.c [hunk 1]
```diff
@@ -15,10 +15,14 @@
   create a matrix of any dimension. The return must be cast correctly.
 */
 void *any_matrix(uint8_t dimension, uint16_t el_size, uint32_t header_size, ...)
-{
+{ 
+    #ifdef _MSC_VER
+        uint16_t *dims = (uint16_t *)_alloca(dimension);
```
Comment (tridge) [inline cuav/image/imageutil.c]: that should be dimension*sizeof(uint16_t)

File: cuav/image/include/imageutil.h [hunk 2]
```diff
@@ -9,11 +9,25 @@
 #include <stdlib.h>
 #include <stdarg.h>
 
-#define PACKED __attribute__((__packed__))
+#ifdef _MSC_VER
```
Comment (tridge) [inline cuav/image/include/imageutil.h:12]: i think we could make macros with the same syntax for both?

File: cuav/image/scanner.c [hunk 3]
```diff
@@ -65,16 +63,29 @@ struct scan_params {
...
-	min_region_area : 8,
-        max_region_area : 400,
-        min_region_size_xy : 2,
-        max_region_size_xy : 30,
-        histogram_count_threshold : 50,
-        region_merge : 1,
-        save_intermediate : false,
-        blue_emphasis : false
-};
+#ifdef _MSC_VER
```
Comment (tridge) [inline cuav/image/scanner.c]: probably better to have just one declaration, with comments on fields

  8, // min_region_area

File: cuav/image/scanner.c [hunk 4]
```diff
@@ -856,7 +867,18 @@ static void mark_regions(struct bgr_image *img, const struct regions *r)
 	}
 }
 
-
+/*
+  map a value based on the threshold
+ */
+uint8_t map_value(float v, const float threshold) {
```
Comment (tridge) [inline cuav/image/scanner.c:860]: should be static and const?

File: cuav/modules/camera_air.py [hunk 5]
```diff
@@ -499,10 +499,10 @@ def start_aircraft_bsend(self):
                 except:
                     print("Bad GCS endpoint (must be remIP:remport:localport:bw): " + str(lnk))
                     pass
-        if self.msend is None:
-            self.msocket = cuav_command.MavSocket(self.mpstate.mav_master[0])
-            self.msend = block_xmit.BlockSender(mss=96, sock=self.msocket, dest_ip='mavlink', dest_port=0, backlog=5, debug=False)
-            self.msend.set_bandwidth(self.camera_settings.m_bandwidth)
+        #if self.msend is None:
```
Comment (tridge) [inline cuav/modules/camera_air.py]: why remove the msend support?

**Review by tridge — CHANGES_REQUESTED** [review]
a few fixes, but overall great work!

File: cuav/modules/camera_air.py (hunk 5 above)
Comment (stephendade) [reply cuav/modules/camera_air.py]: Oops ... re-added that

File: cuav/image/scanner.c (hunk 4 above)
Comment (stephendade) [reply cuav/image/scanner.c:860]: gcc gives compiler errors if I try that -
``error: storage class specified for parameter``

File: cuav/image/scanner.c (hunk 3 above)
Comment (stephendade) [reply cuav/image/scanner.c]: Yep, fixed

File: cuav/image/include/imageutil.h (hunk 2 above)
Comment (stephendade) [reply cuav/image/include/imageutil.h:12]: Couldn't figure out a macro that makes both gcc and Visual Studio happy

File: cuav/image/imageutil.c (hunk 1 above)
Comment (stephendade) [reply cuav/image/imageutil.c]: Tried that, but then it crashes on Windows

**Pull request merged**